dm/relay/streamer.go

```diff
@@ -57,8 +59,18 @@ func (s *LocalStreamer) GetEvent(ctx context.Context) (*replication.BinlogEvent,
 		heartbeatInterval = time.Duration(i) * time.Second
 	})
 
+	fired := false
+	s.heatBeatTimer.Reset(heartbeatInterval)
+	defer func() {
+		if !fired {
```

Comment:
is that mean  in the worst case we need to wait the full 30 seconds before we can exit ?

Reply:
it's a timer for heatbeat event, if worker exit, `ctx.Done`'ll return